cdap-api/src/main/java/co/cask/cdap/api/lineage/LineageGraph.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.lineage;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Graph representing set of {@link Read} operations from the data sources,
+ * set of {@link Transform} operations, and set of {@link Write} operations to the data sinks.
+ */
+public class LineageGraph {
+  private final Set<Read> reads;
+  private final Set<Transform> transforms;
+  private final Set<Write> writes;
+
+  private LineageGraph(Set<Read> reads, Set<Transform> transforms, Set<Write> writes) {
+    this.reads = Collections.unmodifiableSet(new HashSet<>(reads));
+    this.transforms = Collections.unmodifiableSet(new HashSet<>(transforms));
+    this.writes = Collections.unmodifiableSet(new HashSet<>(writes));
+  }
+
+  /**
+   * @return the set of {@link Read} operations in the lineage graph
+   */
+  public Set<Read> getReads() {
+    return reads;
+  }
+
+  /**
+   * @return the set of {@link Transform} operations in the lineage graph
+   */
+  public Set<Transform> getTransforms() {
+    return transforms;
+  }
+
+  /**
+   * @return the set of {@link Write} operations in the lineage graph
+   */
+  public Set<Write> getWrites() {
+    return writes;
+  }
+
+  /**
+   * Builder for building {@link LineageGraph}.
+   */
+  public static class Builder {
+    private final Set<Read> reads;
+    private final Set<Transform> transforms;
+    private final Set<Write> writes;
+
+    public Builder() {
+      this.reads = new HashSet<>();
+      this.transforms = new HashSet<>();
+      this.writes = new HashSet<>();
+    }
+
+    /**
+     * Collect the {@link Read} operations in the lineage graph.
+     * @param reads {@link Read} operation
+     * @return this instance of the Builder
+     */
+    public Builder addRead(Read ... reads) {
```

Comment:
should be addReads (plural) as it takes a number of reads. same for writes and transforms.

Reply:
Removed Builders from all classes based on the discussion with @chtyim, since all the parameters (Read, Write, Transform) are always required.